### DISCUSSION TITLE
Renovate keeps lookup up dependency info from "registry.npmjs.org"

### DISCUSSION THREAD

**Discussion by robg-temp** [renovatebot/renovate#9531] [Request Help]
Hello,  me again!

This is a continuation of [this question](https://github.com/renovatebot/renovate/discussions/9506).  I asked a new question because I have more information now.

We are configuring Renovate to only update internal libraries, which are hosted at:

`https://package.acme.com/repository/npm-private`

It is configured as a self-hosted (Docker) jenkins job that runs every 15 minutes that scans about GitHub 10 projects. We turned on automerging, but it isn't working because of "Artifact update problems", which I believe are caused by renovate looking for updates on the default npm registry:

![image](https://user-images.githubusercontent.com/80468671/114570908-37095600-9c44-11eb-8ed9-d27c8c31b4db.png)

We copy the .npmrc to the root of the docker workspace where the renovate-config.js file also resides:
```
stage('renovate') {
  withEnv([
    "RENOVATE_CONFIG_FILE=$WORKSPACE/renovate-config.js",
    "LOG_LEVEL=info",
    "ACTIVE_BRANCHES=${activeBranches.join(',')}",
  ]) {
    withCredentials([
      // ...
    ]) {
      def r = docker.image('renovate/renovate:24')
      def args = [
        "-v /home/jenkins/.npmrc:$WORKSPACE/.npmrc",
        '-v /home/jenkins/.gradle:/.gradle-user-home:rw,z',
      ].join(' ')
      r.inside(args) {
        sh "renovate --print-config true ${repos.join(' ')}"
      }
    }
  }
}
```

I removed the `"-v /home/jenkins/.npmrc:$WORKSPACE/.npmrc", ` line to see what would happen, and renovate stopped working.  Putting it back made renovate work again, albeit with the 404 errors shown above.

**So my question is, how can I prevent renovate from making calls to "registry.npmjs.org" (or ignore 404 errors) so that automerge works?**

Thanks!

Rob G

**Comment by viceice** [comment]
configure contents of `.npmrc` as `npmrc` renovate option  https://docs.renovatebot.com/private-modules/

Do not copy a `.npmrc` to renovate home, it's deprecated and will be removed in next major coming soon.

**Reply by robg-temp** [reply]
OK, I'll do that.  Thanks!

**Comment by robg-temp** [comment]
I found out what the issue was.  It was caused by the package-lock.json files having outdated references to "registry.npmjs.org".  Even though I removed our internal dependencies, having any references to "registry.npmjs.org" would cause 404 errors for our own libraries. (???)  
Not sure if this is expected behavior or a bug, but, updating ALL references in the package-lock.json file to our private repo OR deleting the package-lock.json file entirely got rid of the 404 errors.

**Reply by viceice** [reply]
This depends on your private npm registry, so are simply redirect to officially npmjs so npm add the public registry url to the lockfile
